app/models/merchant.rb

```diff
@@ -40,4 +40,29 @@ def deactivate_items
   def activate_items
     items.update(active?: true)
   end
+
+  def items_with_default_images
+    default = 'https://static.wixstatic.com/media/d8d60b_6ff8d8667db1462492d681839d85054c~mv2.png/v1/fill/w_900,h_900,al_c,q_90/file.jpg' 
+    items.where(image: default)
+  end
+
+  def unfulfilled_items_total
+    item_orders.where(status: 0).sum('item_orders.price * item_orders.quantity')
+  end
+
+  def unfulfilled_orders_count
+    item_orders.where(status: 0).count
+  end
+
+  def exceeds_inventory?(order)
+    if order.item_orders.joins(:item).where("items.merchant_id = #{self.id} AND item_orders.quantity > items.inventory") != []
+      true
+    else 
+      false
+    end
+  end
+
+  def orders_exceed_inventory?(item)
```

Comment:
this one i used to check against each item in the merchant's view, there might have been an easier way to do it, but I was stumped so went with this way. I check the item's quantity against the sum of the item_orders for that item